src/io.hpp

```
@@ -354,6 +358,16 @@ void read_restart_metadata(parser &user_vals, std::string const &restart_file)
   }
   int const max_level = H5Easy::load<int>(file, std::string("max_level"));
   int const dof       = H5Easy::load<int>(file, std::string("dof"));
+  bool const use_fg   = H5Easy::load<bool>(file, std::string("using_fullgrid"));
+
+  if (use_fg != user_vals.using_full_grid())
+  {
+    throw std::runtime_error("Mismatch of grid type between CLI and restart "
```

Are we sure about this? I mean, whatever grid we read from the file that's what we should use. The only question should be whether we turn adaptivity on/off and even in that case we should use whatever is in the CLI.

The grid that we specify in the CLI should only apply to the initial grid of the problem. What do you guys think?

---

> The grid that we specify in the CLI should only apply to the initial grid of the problem

I agree.

---

Yes, you are right - we only use the grid that is loaded from the file. However, the code does create the grid based on the CLI first, and [then re-creates it if restarting from file](https://github.com/project-asgard/asgard/blob/5715625b08cee81247ce94bcbdfccafad42ece07/src/main.cpp#L213-L224).

This is was meant more as a sanity check to let the user know if the grid options they provide are ignored. For example, if they added a `-f` when loading an adaptive grid, that the `-f` option on the CLI is not used. 

The current behavior is to just silently ignore (minus creating the full grid initially before it gets overwritten by the restart) - we can remove this check if that is the preferred behavior.

---

Remove the check and show a warning, but only if the user explicitly specifies a grid and it is not the one in the file. For example, the default is sparse-grid so if the user just says reload and the file contains a full-grid, then there should be no warning. If the user explicitly says `-f` and the grid is not full, then we can give a warning about ignored CLI parameters.